src/triggers/commentIssueV2.ts

```diff
@@ -0,0 +1,202 @@
+import { omitBy, pick } from "lodash";
+import { ZObject, Bundle } from "zapier-platform-core";
+import sample from "../samples/issueComment.json";
+import { getWebhookData, unsubscribeHook } from "../handleWebhook";
+
+interface Comment {
+  id: string;
+  body: string;
+  url: string;
+  createdAt: string;
+  resolvedAt: string | null;
+  issue: {
+    id: string;
+    identifier: string;
+    title: string;
+    url: string;
+    team: {
+      id: string;
+      name: string;
+    };
+  };
+  user: {
+    id: string;
+    email: string;
+    name: string;
+    avatarUrl: string;
+  };
+  parent: {
+    id: string;
+    body: string;
+    createdAt: string;
+    user: {
+      id: string;
+      email: string;
+      name: string;
+      avatarUrl: string;
+    };
+  } | null;
+}
+
+interface CommentsResponse {
+  data: {
+    comments: {
+      nodes: Comment[];
+    };
+  };
+}
+
+const subscribeHook = (z: ZObject, bundle: Bundle) => {
+  const data = {
+    url: bundle.targetUrl,
+    inputData:
+      bundle.inputData && Object.keys(bundle.inputData).length > 0
+        ? pick(bundle.inputData, ["creatorId", "teamId", "issueId"])
+        : undefined,
+  };
+
+  return z
+    .request({
+      url: "https://client-api.linear.app/connect/zapier/subscribe/commentIssue",
+      method: "POST",
+      body: data,
+    })
+    .then((response) => response.data);
+};
+
+const getCommentList = () => async (z: ZObject, bundle: Bundle) => {
+  const variables = omitBy(
+    {
+      creatorId: bundle.inputData.creatorId,
+      teamId: bundle.inputData.teamId,
+      issueId: bundle.inputData.issueId,
+    },
+    (v) => v === undefined
+  );
+
+  const filters = [`{ issue: { null: false } }`];
+  if ("creatorId" in variables) {
+    filters.push(`{ user: { id: { eq: $creatorId } } }`);
+  }
+  if ("teamId" in variables) {
+    filters.push(`{ issue: { team: { id: { eq: $teamId } } } }`);
+  }
+  if ("issueId" in variables) {
+    filters.push(`{ issue: { id: { eq: $issueId } } }`);
+  }
+
+  const response = await z.request({
+    url: "https://api.linear.app/graphql",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: bundle.authData.api_key,
+    },
+    body: {
+      query: `
+      query ZapierListCommentsV2${
+        Object.keys(variables).length === 0
+          ? ""
+          : `(
+        ${"creatorId" in variables ? "$creatorId: ID" : ""}
+        ${"teamId" in variables ? "$teamId: ID" : ""}
+        ${"issueId" in variables ? "$issueId: ID" : ""}
+      )`
+      } {
+        comments(
+          first: 25
+          ${
+            filters.length > 0
+              ? `
+          filter: {
```

Comment:
I know this was already here, so we're kind of working off that but I wonder could we use our own SDK for this? 
Short of that could we use a GQL query builder?
All of this string manipulation kind of sucks

Reply:
Let's go the GQL query builder route - it'll be harder to test this against your local API if we move to the SDK.

Reply:
Going to merge this as it is now and open another PR to move the new triggers to use a query builder